javaslang/src/main/java/javaslang/collection/Arrays.java

```diff
@@ -27,6 +27,30 @@
         return arrayCopy(length, array, from, 0, length);
     }
 
+    /** Repeatedly group an array into equal sized sub-trees */
+    static Object[] grouped(Object[] array, int groupSize) {
+        assert array.length > groupSize;
+        final Object[] results = copy(empty(), 1 + ((array.length - 1) / groupSize));
+        results[0] = copyRange(array, 0, groupSize);
+
+        for (int start = groupSize, i = 1; start < array.length; i++) {
+            final int nextLength = Math.min(groupSize, array.length - (i * groupSize));
+            results[i] = copyRange(array, start, start + nextLength);
+            start += nextLength;
+        }
+
+        return results;
+    }
+
+    /** Store the content of an iterable in an array */
+    static Object[] asArray(java.util.Iterator<?> it, int length) {
```

Comment:
We could change it to use `Iterable`:

```java
static Object[] asArray(java.util.Iterable<?> it, int length)
```

That saves us `iterator()` calls within our code-base, right? I can't find here more uses of this method but maybe there exist already other uses where we only have an Iterator at hand instead of an Iterable - just a question.